modules/core/shared/src/main/scala/io/circe/Encoder.scala

```diff
@@ -88,6 +88,10 @@ trait Encoder[A] extends Serializable { self =>
   final def mapJson(f: Json => Json): Encoder[A] = new Encoder[A] {
     final def apply(a: A): Json = f(self(a))
   }
+
+  def at(field: String): Encoder[A] = new Encoder[A] {
```

Comment:
I think this is a reasonable ask, but can you add some scaladoc, and tests for both this at and the at in Codec?  Make sure to add some law tests to make sure an at'd Codec obeys the codec laws.